src/Resources/contao/models/PageModel.php

```diff
@@ -433,6 +433,34 @@ public static function findFirstPublishedRegularByPid($intPid, array $arrOptions
 	}
 
 
+	/**
+	 * Find an error 401 page by its parent ID
+	 *
+	 * @param integer $intPid     The parent page's ID
+	 * @param array   $arrOptions An optional options array
+	 *
+	 * @return PageModel|null The model or null if there is no 401 page
+	 */
+	public static function find401ByPid($intPid, array $arrOptions=array())
+	{
+		$t = static::$strTable;
+		$arrColumns = array("$t.pid=? AND $t.type='error_401'");
```

Comment:
so this only works if the page is a direct child of the root page? Is this already the case for 403?

Reply:
Yes.